packages/notebook/src/widget.ts

```diff
@@ -1154,6 +1165,10 @@ class Notebook extends StaticNotebook {
   protected onAfterAttach(msg: Message): void {
     super.onAfterAttach(msg);
     let node = this.node;
+    // We must use to capture phase to prevent the code editor
+    // from receiving the mousedown so we can intercept right click
+    // event.
+    node.addEventListener('mousedown', this, true);
     node.addEventListener('mousedown', this);
```

Comment:
Is this double-mousdown-event-handling intended? It seems to be generating a lot of focus noise in my testing.

Reply:
I have a commit or two to push that change some of the logic to make it clearer, I think.